### PR TITLE
feat(nav): Ecosystem + Solutions navigation menu

### PR DESCRIPTION
**Pull request by charlypoly**
- [x] Ecosystem menu
- [x] Solutions menu

### CODE REVIEW

File: packages/components/src/components/SolutionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  List,
+} from './SolutionsMenu.styles';
+
+const categories: { title: string; items: { title: string; url: string }[] }[] =
+  [
+    {
+      title: 'Guides',
+      items: [
+        {
+          title: 'GraphQL Error handling',
```
Comment (n1ru4l) [inline packages/components/src/components/SolutionsMenu.tsx:16]:
```suggestion
          title: 'GraphQL error handling',
```

Comment (n1ru4l) [reply packages/components/src/components/SolutionsMenu.tsx:16]:
```suggestion
          title: 'GraphQL Error Handling',
```

Comment (n1ru4l) [reply packages/components/src/components/SolutionsMenu.tsx:16]: either all title case or all lower-case

File: packages/components/src/components/SolutionsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          url: 'https://www.the-guild.dev/blog/graphql-error-handling-with-fp',
+        },
+        {
+          title: 'GraphQL Authentication',
```
Comment (n1ru4l) [inline packages/components/src/components/SolutionsMenu.tsx:20]:
```suggestion
          title: 'GraphQL authentication',
```

File: packages/components/src/components/SolutionsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          url: 'https://www.envelop.dev/docs/guides/securing-your-graphql-api',
+        },
+        {
+          title: 'Monitor your GraphQL API',
+          url: 'https://www.envelop.dev/docs/guides/monitoring-and-tracing',
+        },
+      ],
+    },
+    {
+      title: 'GraphQL at scale',
```
Comment (n1ru4l) [inline packages/components/src/components/SolutionsMenu.tsx:54]:
```suggestion
      title: 'GraphQL at Scale',
```

File: packages/components/src/components/SolutionsMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      items: [
+        {
+          title: 'Schema Stitching',
+          url: 'https://www.the-guild.dev/blog/a-new-year-for-schema-stitching',
+        },
+        {
+          title: 'Manage your Schemas',
```
Comment (n1ru4l) [inline packages/components/src/components/SolutionsMenu.tsx:61]:
```suggestion
          title: 'Manage your schemas',
```

**Review by notrab — APPROVED** [review]
Nailed it! :shipit:

File: packages/components/src/components/Ecosystem.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  () => css`
+    ${tw`relative mr-3`}
+    img {
+      ${tw`max-w-[25px] align-bottom last:hidden`}
+    }
+  `
+);
+
+export const ProductThumbnail = styled.a(() => [
+  tw`box-border flex items-center w-full my-1 md:(w-1/2 m-0) py-2 px-2 rounded-lg outline-none no-underline!`,
```
Comment (B2o5T) [inline packages/components/src/components/Ecosystem.styles.tsx:24]:
```suggestion
  tw`box-border flex items-center w-full my-1 md:(w-1/2 m-0) p-2 rounded-lg outline-none no-underline!`,
```

File: packages/components/src/components/Ecosystem.styles.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  tw`dark:bg-gray-800 bg-white`,
+  {
+    maxHeight: 'calc(100vh - 70px)',
+    overflowY: 'scroll',
+    padding: '20px',
+    minHeight: '300px',
+    width: '800px',
+    borderRadius: 6,
+    boxShadow:
+      'hsl(206 22% 7% / 35%) 0px 10px 38px -10px, hsl(206 22% 7% / 20%) 0px 10px 20px -15px',
```
Comment (B2o5T) [inline packages/components/src/components/Ecosystem.styles.tsx:54]: some of these property can be specified with tailwindcss but it's fine, we'll do refactor later

File: packages/components/src/components/Header.tsx [hunk 7]
```diff
@@ -28,14 +38,19 @@ export const Header: React.FC<IHeaderProps> = ({
...
+  const { height: windowHeight, width: windowWidth } = useWindowSize();
 
-  const handleModal = (state: boolean) => {
-    !mobileNavOpen && toggleLockBodyScroll(state);
-    setModalOpen(state);
-  };
+  const shouldUseMenu = useMemo(
+    () =>
+      windowWidth &&
+      windowWidth &&
```
[the remark this answers is not in the document]
Comment (B2o5T) [reply packages/components/src/components/Header.tsx:49]: I wanted to say that you duplicated `windowWidth &&`
```suggestion
```

File: packages/components/src/components/SolutionsMenu.styles.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    h3 {
+      ${tw`w-full mt-1 mb-5 font-normal text-base text-gray-600 dark:text-gray-400 text-black`}
+    }
+  `,
+]);
+
+export const List = styled.div(() => [tw`flex flex-wrap`]);
+
+export const Item = styled.a(() => [
+  tw`box-border flex items-center w-full my-1 md:(w-1/2 m-0) py-2 px-2 rounded-lg outline-none no-underline!`,
```
Comment (B2o5T) [inline packages/components/src/components/SolutionsMenu.styles.tsx:15]:
```suggestion
  tw`box-border flex items-center w-full my-1 md:(w-1/2 m-0) p-2 rounded-lg outline-none no-underline!`,
```

File: packages/components/src/components/SolutionsMenu.styles.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      }
+    }
+  `,
+]);
+
+export const Callout = styled.div(() => [
+  tw`mt-10 p-2`,
+  css`
+    display: flex;
+    justify-content: end;
```
Comment (B2o5T) [inline packages/components/src/components/SolutionsMenu.styles.tsx:40]:
```suggestion
  tw`mt-10 p-2 flex justify-end`,
  css`
```
also `justify-content: end` don't exist
image

**Comment by B2o5T** [comment]
I think bg color on dark mode must be more darker, similar to codegen or website 
image
image
image

**Comment by B2o5T** [comment]
I think headings should be aligned with links, or links should not have padding-x
image

**Comment by B2o5T** [comment]
we can add arrowdown to both navigation menu that will be animated, rotated for 180deg on hover
image

**Comment by B2o5T** [comment]
I think you can reduce space before last link write your... 
image

**Review by B2o5T — APPROVED** [review]

**Pull request merged**